### ISSUE TITLE
Add Python 3.10 Support

### ISSUE THREAD

**Issue by Martin-Molinero** [QuantConnect/lean-cli#54]

**Comment by JakeTheSnake3p0** [comment]
I'm in need of this upgrade as libraries like tensorflow cannot be upgraded without it. As a consequence, I cannot use features of tensorflow-probability due to an old bug.

Is it possible for users to specify a version of python they'd like to use through the CLI?

**Comment by Martin-Molinero** [comment]
Update: docker basically doesn't work at the moment with python 3.10 https://github.com/docker/docker-py/issues/2902#issuecomment-1104141940

**Comment by JakeTheSnake3p0** [comment]
@Martin-Molinero looks like there's been some progress on that.

**Comment by Martin-Molinero** [comment]
> @Martin-Molinero looks like there's been some progress on that.

yes, thank you for the heads up! They haven't released a new version to pip yet, probably will happen in the next few days 🤞

**Comment by JakeTheSnake3p0** [comment]
Any word on research environment upgrade to 3.10? As of today it's still reporting 3.8.13.
